space_packet_parser/parameters.py

```diff
@@ -312,15 +310,13 @@ def parse_value(self, packet: packets.CCSDSPacket, **kwargs):
 
         Returns
         -------
-        parsed_value : int
-            Raw encoded value
-        derived_value : str
+        derived_value : BoolParameter
             Resulting boolean representation of the encoded raw value
         """
-        raw, _ = super().parse_value(packet, **kwargs)
+        raw = super().parse_value(packet, **kwargs)
         # Note: This behaves very strangely for String and Binary data encodings.
         # Don't use those for Boolean parameters. The behavior isn't specified well in XTCE.
-        return raw, bool(raw)
+        return packets.BoolParameter(bool(raw), raw)
```

Comment:
@medley56 I did have to update a test for this which I wanted to get your take on.

The `raw` here is actually the "derived value" since it is the transformed return. So we are doing `bool(derived_value)` now. Is that what we want, or do we want to have it as:
```python
raw = super().parse_value(packet, **kwargs).raw_value
```

It seems like you'd want the derived value to produce the boolean of? i.e. if I have a calibrated value that is `raw + 1` and my raw value is `0`. I'd want that to be True and not False.

1. `bool(value), value`: current implementation, tells a user what was converted to the boolean
2. `bool(value), value.raw_value`: gives the user the most "raw" representation, but avoids the intermediary encoding
3. `bool(value.raw_value), value.raw_value`: only ever uses the raw value and avoids any derived/calibrated values.

Which version should we use here?

Reply:
Oof, this is a good question. I think we want BooleanParameterType to take the bool of the derived value. The issue is that we have discovered a parameter type that has multiple intermediate representations from raw data --> raw value --> calibrated value --> bool. This is something I never noticed but now I wonder if we have the same problem for enumerated parameter types. 

Let's skip the calibrated (intermediate) value as you suggest in option 2.

Reply:
OK, I just pushed (2) up.

Re: Enumerated... We are only passing back StringParameters there now. Is there a way to have something else in XTCE like pointing to a different parameter lookup or to say "read this list as an int"? Something we can tackle in a follow-up issue and would be a bug-fix if anything.

Reply:
#93 Will address this.